lib/importproject.cpp

```diff
@@ -194,28 +194,33 @@ ImportProject::Type ImportProject::import(const std::string &filename, Settings
     if (endsWith(filename, ".json")) {
         if (importCompileCommands(fin)) {
             setRelativePaths(filename);
+            setFileSizes();
```

Comment:
This block should be generic to avoid redundancies. Does not need to be addressed in this PR (I might possibly tackle it in my changes).

Reply:
This should be unnecessary,

The sizes should be updated in `CmdLineParser::fillSettingsFromArgs()` so it is in the common place. Similar to what is done in #7116 we should do this for all entries which have 0 as size.

Reply:
We also do not need to bother with not having redundant code there. that will go away in the future having all the logic inline will make it easier to review it for differences in the future when this actually gets merged.